### ISSUE TITLE
[java] False positive FinalFieldCouldBeStatic when using lombok Builder.Default

### ISSUE THREAD

**Issue by ollieabbey** [pmd/pmd#2708]
**Affects PMD Version:** 

maven-pmd-plugin version 3.13.0 ([PMD 6.21.0](https://maven.apache.org/plugins/maven-pmd-plugin/examples/upgrading-PMD-at-runtime.html#Version_relationship_between_maven-pmd-plugin_and_PMD))

**Rule:** [FinalFieldCouldBeStatic](https://pmd.github.io/latest/pmd_rules_java_design.html#finalfieldcouldbestatic)

**Description:**

Private final fields annotated with [lombok](https://projectlombok.org/features/Builder) `@Builder.Default` trigger FinalFieldCouldBeStatic, even though they're not behaving as constants.

**Code Sample demonstrating the issue:**

```java
package com.example;

import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class ExampleClass {

    @Builder.Default
    private final long exampleField = 0L;
}
```

**Output**

`PMD Failure: com.example.ExampleClass:11 Rule:FinalFieldCouldBeStatic Priority:3 This final field could be made static.`

**Running PMD through:** *[Maven]*

**Comment by ollieabbey** [comment]
I've created https://github.com/pmd/pmd/pull/2747 to fix this.

 @adangel if you wouldn't mind reviewing and letting me know if there's anything that needs changing, I can do so. Cheers!
